src/main/scala/DataBaseConn.scala

```diff
@@ -1,14 +1,67 @@
 import DataBaseConn.xa
-import cats.effect.IO
+import cats.effect.{Blocker, IO}
+import com.zaxxer.hikari.HikariDataSource
 import doobie._
+import doobie.hikari.HikariTransactor
 import doobie.implicits._
+import doobie.implicits.javasql._
+
+import java.sql.Timestamp
 import scala.concurrent.ExecutionContext
 
+trait UrlDatabaseFunctions {
```

Comment:
I would call it `UrlRepository`. For the reason, check https://docs.microsoft.com/en-us/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/infrastructure-persistence-layer-design